bandplotter.py

```diff
@@ -209,9 +240,27 @@ def plot_bands(
             self._last_color = kwargs['c']
         else:
             self._last_color = kwargs['color']
-
-        self._ax.plot(banddata, formatstr, label=label, **kwargs)
-
+        
+        if correct_x_axis:
+            x_vals = self._calc_corrected_x_values(k_data)
+            self._ax.plot(x_vals, banddata, formatstr, label=label, **kwargs)
+            
+            # TODO: This is an example on how to use scatter to plot the bands.
+            # This will be needed to color the points by e.g. parity. To do
+            # sp simply use the `c`-argument of scatter.
+#            for bd in banddata.T:
+#                self._ax.scatter(x_vals, bd, label=label, zorder=1000, 
+#                                 **kwargs)
+            
+            # we need to update the reference to the x_data and the 
+            # x_axis_formatter ticks accordingly to get the lightcone and the
+            # x-axis labels right
+            # TODO: needs to be checked if it works in all cases
+            self._x_data = x_vals
+            x_axis_formatter._ticks = x_vals[x_axis_formatter._ticks]
```

Comment:
I have to think how&where to properly implement this in x_axis_formatter. I don't like setting the "private" attribute _ticks here. But the resulting plot looks good, so it works so far, so I think we can leave it like this for now. We should at least add a # TODO for this as well ;).